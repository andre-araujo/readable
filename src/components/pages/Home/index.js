import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import * as actions from '../../../actions';

import PostList from '../../modules/PostList';
import CategoryNav from '../../modules/CategoryNav';

class Home extends Component {
  componentDidMount() {
    this.props.getCategories();
  }

  render() {
    const {
      categories
    } = this.props

    return (
      <div>
        <CategoryNav categories={categories} />
        <PostList />
      </div>
    );
  }
}

const mapStateToProps = ({ categories }) => ({
  categories: categories.categories
})

const mapDispatchToProps = (dispatch) => ({
  getCategories() {
    dispatch(actions.getCategories())
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Home);
