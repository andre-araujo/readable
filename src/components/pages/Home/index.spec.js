import React from 'react';
import { shallow } from 'enzyme';

import Home from '.';

describe('Home', () => {
  it('should match snapshot', () => {
    expect(shallow(<Home />)).toMatchSnapshot();
  });
});
