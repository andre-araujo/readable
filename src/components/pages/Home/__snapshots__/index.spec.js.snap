// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <h1>
    Home
  </h1>
</div>
`;
